exotations/collision_scene_fcl_latest/src/CollisionSceneFCLLatest.cpp

```diff
@@ -0,0 +1,381 @@
+        const std::string& name2 = e2->ClosestRobotLink?e2->ClosestRobotLink->Segment.getName():e2->Parent->Segment.getName();
+        return scene->acm_.getAllowedCollision(name1, name2);
+    }
+    return true;
+}
+
+void CollisionSceneFCLLatest::checkCollision(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, CollisionData* data)
+{
+    data->Request.num_max_contacts = 1000;
+    data->Result.clear();
+    fcl::collide(o1,o2,data->Request, data->Result);
+}
+
+bool CollisionSceneFCLLatest::collisionCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void* data)
+{
+    CollisionData* data_ = reinterpret_cast<CollisionData*>(data);
+
+    if(!isAllowedToCollide(o1, o2, data_->Self, data_->Scene)) return false;
+
+    checkCollision(o1, o2, data_);
+    return data_->Result.isCollision();
+}
+
+void CollisionSceneFCLLatest::computeDistance(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, DistanceData* data)
+{
+    data->Request.enable_nearest_points = true;
+    data->Request.enable_signed_distance = true; // Added in FCL 0.6.0
+    // GST_LIBCCD produces incorrect contacts. Probably due to incompatible version of libccd.
+    data->Request.gjk_solver_type = fcl::GST_INDEP;
+    data->Result.clear();
+    fcl::distance(o1,o2,data->Request, data->Result);
+
+    CollisionProxy p;
+    p.e1 = reinterpret_cast<KinematicElement*>(o1->getUserData());
+    p.e2 = reinterpret_cast<KinematicElement*>(o2->getUserData());
+
+    p.distance = data->Result.min_distance;
+    if(p.distance>0)
+    {
+        KDL::Vector c1 = p.e1->Frame*KDL::Vector(data->Result.nearest_points[0](0), data->Result.nearest_points[0](1), data->Result.nearest_points[0](2));
+        KDL::Vector c2 = p.e2->Frame*KDL::Vector(data->Result.nearest_points[1](0), data->Result.nearest_points[1](1), data->Result.nearest_points[1](2));
+        KDL::Vector n1 = c2-c1;
+        KDL::Vector n2 = c1-c2;
+        n1.Normalize();
+        n2.Normalize();
+        tf::vectorKDLToEigen(c1, p.contact1);
+        tf::vectorKDLToEigen(c2, p.contact2);
+        tf::vectorKDLToEigen(n1, p.normal1);
+        tf::vectorKDLToEigen(n2, p.normal2);
+    }
+    else
+    {
+        // Contact points are not reliable (FCL bug), use shape centres instead.
+        KDL::Vector c1 = p.e1->Frame.p;
+        KDL::Vector c2 = p.e2->Frame.p;
+        KDL::Vector n1 = c2-c1;
+        KDL::Vector n2 = c1-c2;
+        n1.Normalize();
+        n2.Normalize();
+        tf::vectorKDLToEigen(c1, p.contact1);
+        tf::vectorKDLToEigen(c2, p.contact2);
+        tf::vectorKDLToEigen(n1, p.normal1);
+        tf::vectorKDLToEigen(n2, p.normal2);
+    }
+    data->Distance = std::min(data->Distance, p.distance);
+    data->Proxies.push_back(p);
+}
+
+bool CollisionSceneFCLLatest::collisionCallbackDistance(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void* data, double& dist)
```

Comment:
``dist`` is now being passed in as a reference but not assigned a value/used as part of the function - computeDistance changes the values in ``data_`` but ``dist`` does not get assigned a value.